api/src/main/java/jakarta/servlet/http/HttpUtils.java

```diff
@@ -1,249 +0,0 @@
-/*
- * Copyright (c) 1997, 2020 Oracle and/or its affiliates and others.
- * All rights reserved.
- * Copyright 2004 The Apache Software Foundation
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package jakarta.servlet.http;
-
-import jakarta.servlet.ServletInputStream;
-import java.io.IOException;
-import java.util.Hashtable;
-import java.util.ResourceBundle;
-import java.util.StringTokenizer;
-
-/**
- * @deprecated As of Java(tm) Servlet API 2.3. These methods were only useful with the default encoding and have been
- * moved to the request interfaces.
- *
- */
-@Deprecated
-public class HttpUtils {
-
-    private static final String LSTRING_FILE = "jakarta.servlet.http.LocalStrings";
-    private static ResourceBundle lStrings = ResourceBundle.getBundle(LSTRING_FILE);
-
-    /**
-     * Constructs an empty <code>HttpUtils</code> object.
-     */
-    public HttpUtils() {
-    }
-
-    /**
-     * Parses a query string passed from the client to the server and builds a <code>HashTable</code> object with key-value
-     * pairs. The query string should be in the form of a string packaged by the GET or POST method, that is, it should have
-     * key-value pairs in the form <i>key=value</i>, with each pair separated from the next by a &amp; character.
-     *
-     * <p>
-     * A key can appear more than once in the query string with different values. However, the key appears only once in the
-     * hashtable, with its value being an array of strings containing the multiple values sent by the query string.
-     * 
-     * <p>
-     * The keys and values in the hashtable are stored in their decoded form, so any + characters are converted to spaces,
-     * and characters sent in hexadecimal notation (like <i>%xx</i>) are converted to ASCII characters.
-     *
-     * @param s a string containing the query to be parsed
-     *
-     * @return a <code>HashTable</code> object built from the parsed key-value pairs
-     *
-     * @exception IllegalArgumentException if the query string is invalid
-     */
-    public static Hashtable<String, String[]> parseQueryString(String s) {
```

Comment:
@markt-asf Could you suggest replacement for this method, please?

Reply:
`ServletRequest.getParameterMap()`. If an application needs to differentiate between query string parameters and request body parameters it will need to implement code to do that itself. I suspect parsing the query string would be the simplest to implement.

Reply:
@markt-asf - thank you for confirmation. I was exactly thinking about body params being included in `getParameterMap`. It's clear now.